confd/pkg/backends/calico/client.go

```diff
@@ -538,12 +539,23 @@ func (c *client) updatePeersV1() {
 					}
 				}
 
+				// Check if the peer represents a node Calico is running on.
+				var isCalicoNode bool
+				for _, nodeName := range c.nodeLabelManager.listNodes() {
```

Comment:
Maybe not a real concern, but the label manager only stores nodes that have labels attached to them IIRC. I think if a node is added and has no labels, it might not show up here?

Also, this is going to iterate the nodes a couple of times for each BGP peer - at a minimum we should adjust this so we only call listNodes once per sync, maybe we should be keeping a map of IP address -> isCalicoNode so we can do the lookup for basically free.

Reply:
I contemplated doing it that way too, but then my mapping would have the IPs as the key and probably the node name as the value so that I could delete the IP if the node is removed. In that case I would be looping over my set of nodes every time a node is removed (which is only referenced by name) which I figured was worse to loop over everything when we didn't have to as opposed to adding another loop when we were already going to loop over everything. I can see about mushing all of this logic inside of where we normally loop over this which would be to retrieve the node name so at least the loop can be done once though.

The label manager also stores the nodes even if there are no labels attached to them (maybe incorrectly?) https://github.com/projectcalico/calico/blob/master/confd/pkg/backends/calico/node_labels.go#L48 since we just pass in whatever the node's labels are without inspecting the set so with the logic how it is now, it should hold all the nodes.

Reply:
actually scrap it all, I had overlooked using the existing cache properly to make a node IP cache easy to maintain. I'll update with a node IP cache since that'll just be better.